src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/OneTimeLinkController.kt

```diff
@@ -18,9 +15,20 @@ class OneTimeLinkController(
         val context: ControllerContext,
         val controllers: MontaguControllers,
         private val errorHandler: ErrorHandler = ErrorHandler(),
-        private val redirectValidator: RedirectValidator = MontaguRedirectValidator()
+        private val redirectValidator: RedirectValidator = MontaguRedirectValidator(),
+        onetimeLinkResolver: OnetimeLinkResolver? = null
 ) : AbstractController(context)
 {
+
+    private val onetimeLinkResolver = if (onetimeLinkResolver == null)
+    {
+        OnetimeLinkResolver(controllers, this.repos)
```

Comment:
Why not just make this a default parameter value? Or failing that, you could use an elvis operator (`?:`)

Reply:
Can't be a default param because this.repos isn't available in the constructor. Will change to elvis though